apps/prairielearn/src/pages/studentAssessments/studentAssessments.html.ts

```diff
@@ -0,0 +1,188 @@
+                              <a
+                                href="${urlPrefix}${row.link}"
+                                class="badge color-${row.assessment_set_color} color-hover"
+                                data-testid="assessment-set-badge"
+                              >
+                                ${row.label}
+                              </a>
+                            `}
+                      </td>
+                      <td class="align-middle">
+                        ${row.multiple_instance_header ||
+                        (!row.active && row.assessment_instance_id == null)
+                          ? row.title
+                          : html`
+                              <a href="${urlPrefix}${row.link}">
+                                ${row.title}
+                                ${row.group_work
+                                  ? html`<i class="fas fa-users" aria-hidden="true"></i>`
+                                  : ''}
+                              </a>
+                            `}
+                      </td>
+                      <td class="text-center align-middle">
+                        ${row.assessment_instance_open !== false
+                          ? row.credit_date_string
+                          : 'Assessment closed.'}
+                        ${renderEjs(
+                          import.meta.url,
+                          "<%- include('../partials/studentAccessRulesPopover'); %>",
+                          {
+                            accessRules: row.access_rules,
+                            assessmentSetName: row.assessment_set_name,
+                            assessmentNumber: row.assessment_number,
+                          },
+                        )}
+                      </td>
+                      <td class="text-center align-middle">
+                        ${row.multiple_instance_header
+                          ? row.active
+                            ? html`
+                                <a href="${urlPrefix}${row.link}" class="btn btn-primary btn-sm">
+                                  New instance
+                                </a>
+                              `
+                            : html`
+                                <button type="button" disabled class="btn btn-primary btn-sm">
+                                  New instance
+                                </button>
+                              `
+                          : row.assessment_instance_id
+                            ? row.show_closed_assessment_score
+                              ? renderEjs(
+                                  import.meta.url,
+                                  "<%- include('../partials/scorebar'); %>",
+                                  { score: row.assessment_instance_score_perc },
+                                )
+                              : 'Score not shown'
+                            : 'Not started'}
```

Comment:
This is a pretty gross nested ternary. Consider moving to a separate component with early returns:

```ts
function NameThisThingSomething({ row }) {
  if (row.multiple_instance_header) {
    if (row.active) {
      return 'New instance HTML goes here';
    }

    return 'Disabled HTML';
  }

  if (...) ...
}
```

Reply:
Reorganized in 8b6b7af25.